### DISCUSSION TITLE
[@types/video.js] Should getComponent method use generic type?

### DISCUSSION THREAD

**Discussion by nonth** [DefinitelyTyped/DefinitelyTyped#61426] [Issues with a @types package]
Hi 👋

According to video.js feature that allow us to create our own Component I think change `getComponent` method in Component to use generic type like below will make more efficient than hardcode type in current  type definitions. 

```
getComponent<T>(name: string): T;
```

then we can use

```ts
const SeekBar = videojs.getComponent<videojs.SeekBar>('SeekBar');
```

What do you think?

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
IMO it cannot be just a simple generic method. It would require some conditional types implementation, as the current TS support is a really low friction one for JS users. Custom types require just casting to proper type, there is no requirement to specify type by default, just provide proper string token, as in plain JS code.
